From e61ece2b42b270dd3f68718fef291be7b7f44aa6 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Tue, 2 Nov 2010 23:19:07 +0100
Subject: [PATCH 1/4] make custom session work

Gentoo bug: #216984

fix custom sessions not doing sourcing in the proper order.
---
 data/Xsession.in | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/data/Xsession.in b/data/Xsession.in
index 118518c..201be92 100755
--- a/data/Xsession.in
+++ b/data/Xsession.in
@@ -155,15 +155,6 @@ fi
 
 xhost +si:localuser:`id -un` || :
 
-# run all system xinitrc shell scripts.
-if [ -d /etc/X11/xinit/xinitrc.d ]; then
-    for i in /etc/X11/xinit/xinitrc.d/* ; do
-        if [ -x "$i" -a ! -d "$i" ]; then
-	    . "$i"
-        fi
-    done
-fi
-
 if [ "x$command" = "xcustom" ] ; then
   if [ -x "$HOME/.xsession" ]; then
     command="$HOME/.xsession"
@@ -191,6 +182,15 @@ if [ "x$command" = "xdefault" ] ; then
   fi
 fi
 
+# run all system xinitrc shell scripts.
+if [ -d /etc/X11/xinit/xinitrc.d ]; then
+    for i in /etc/X11/xinit/xinitrc.d/* ; do
+        if [ -x "$i" -a ! -d "$i" ]; then
+	    . "$i"
+        fi
+    done
+fi
+
 # add ssh-agent if found
 sshagent="`gdmwhich ssh-agent`"
 if [ -n "$sshagent" ] && [ -x "$sshagent" ] && [ -z "$SSH_AUTH_SOCK" ]; then
-- 
1.8.5.1

